src/app/products/components/cart-tooltip/cart-tooltip.component.ts

```diff
@@ -0,0 +1,35 @@
+import { Component, Input } from '@angular/core';
+import { retryWhen } from 'rxjs';
+import Product from '../../interfaces/product.interface';
+import { CartService } from '../../services/cart.service';
+
+@Component({
+  selector: 'app-cart-tooltip',
+  templateUrl: './cart-tooltip.component.html',
+  styleUrls: ['./cart-tooltip.component.scss'],
+})
+export class CartTooltipComponent {
+  items: Map<Product, number>;
+  visibility: boolean;
+
+  constructor(private cartService: CartService) {}
+
+  public deleteItem(item: Product): void {
+    this.cartService.removeFromCart(item);
+  }
+
+  public getTotalPriceOfProducts(): number {
```

Comment:
Look, you are doing the same calculations in two places. Here and in the cart component. Better to mave this logic to service